FlowCrypt/src/androidTest/java/com/flowcrypt/email/ui/base/BaseDraftsGmailAPIFlowTest.kt

```diff
@@ -0,0 +1,397 @@
+      .perform(open())
+
+    onView(withId(R.id.navigationView))
+      .perform(clickOnFolderWithName(JavaEmailConstants.FOLDER_DRAFT))
+
+    Thread.sleep(1000)
+  }
+
+  protected fun handleCommonAPICalls(request: RecordedRequest): MockResponse {
+    when {
+      request.path == "/v1/keys/private" -> {
+        return MockResponse().setResponseCode(HttpURLConnection.HTTP_OK).setBody(
+          ApiHelper.getInstance(getTargetContext()).gson
+            .toJson(EkmPrivateKeysResponse(privateKeys = listOf(Key(decryptedPrivateKey))))
+        )
+      }
+
+      request.path == "/gmail/v1/users/me/settings/sendAs" -> {
+        return MockResponse().setResponseCode(HttpURLConnection.HTTP_OK).setBody(
+          ListSendAsResponse().apply {
+            factory = GsonFactory.getDefaultInstance()
+            sendAs = emptyList()
+          }.toString()
+        )
+      }
+
+      request.path == "/gmail/v1/users/me/labels" -> {
+        return MockResponse().setResponseCode(HttpURLConnection.HTTP_OK).setBody(
+          ListLabelsResponse().apply {
+            factory = GsonFactory.getDefaultInstance()
+            labels = addLabelsToDatabaseRule.folders.map {
+              Label().apply {
+                id = it.fullName
+                name = it.folderAlias
+              }
+            }
+          }.toString()
+        )
+      }
+
+      request.path == "/gmail/v1/users/me/messages?labelIds=${JavaEmailConstants.FOLDER_INBOX}&maxResults=45" -> {
+        return MockResponse().setResponseCode(HttpURLConnection.HTTP_OK).setBody(
+          ListMessagesResponse().apply {
+            factory = GsonFactory.getDefaultInstance()
+            messages = emptyList()
+          }.toString()
+        )
+      }
+
+      request.method == "GET" && request.path == "/gmail/v1/users/me/drafts?maxResults=45" -> {
+        return MockResponse().setResponseCode(HttpURLConnection.HTTP_OK).setBody(
+          ListDraftsResponse().apply {
+            factory = GsonFactory.getDefaultInstance()
+            drafts = draftsCache.map { draft ->
+              Draft().apply {
+                id = draft.id
+                message = Message().apply {
+                  id = draft.message.id
+                  threadId = draft.message.threadId
+                }
+              }
+            }
+          }.toString()
+        )
+      }
+
+      request.method == "GET" && request.path == "/gmail/v1/users/me/drafts?fields=drafts/id,drafts/message/id&maxResults=500" -> {
+        return MockResponse().setResponseCode(HttpURLConnection.HTTP_OK).setBody(
+          ListDraftsResponse().apply {
+            factory = GsonFactory.getDefaultInstance()
+            drafts = draftsCache.map { draft ->
+              Draft().apply {
+                id = draft.id
+                message = Message().apply {
+                  id = draft.message.id
+                }
+              }
+            }
+          }.toString()
+        )
+      }
```

Comment:
I think we can combine those 2 handlers and add add condition inside combined handler?